go/controller/cmd/main.go

```diff
@@ -215,6 +215,9 @@ func main() {
 		})
 	}
 
+	// filter out invalid namespaces from the watchNamespaces flag (comma separated list)
+	watchNamespacesList := filterValidNamespaces(strings.Split(watchNamespaces, ","))
```

Comment:
separated this logic out since the validated namespaces would be used for both the cache option + passing to the namespace handler (where we list namespaces from)